### PR TITLE
Use a unique key for the accounts

### PR DESCRIPTION
**Pull request by aaomidi**
This PR fixes https://github.com/microsoft/azuredatastudio/issues/11353

### CODE REVIEW

File: extensions/mssql/src/features.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ export class AccountFeature implements StaticFeature {
 
 		constructor(account: azdata.Account) {
 			this.account = account;
-			this.label = account.key.accountId;
+			this.label = `${account.displayInfo.displayName} - ${account.displayInfo.email ?? ''}`;
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/features.ts:104]: Don't have the - if there's no e-mail

File: src/sql/workbench/services/connection/browser/connectionWidget.ts [hunk 2]
```diff
@@ -520,7 +520,7 @@ export class ConnectionWidget extends lifecycle.Disposable {
 		let oldSelection = this._azureAccountDropdown.value;
 		const accounts = await this._accountManagementService.getAccounts();
 		this._azureAccountList = accounts.filter(a => a.key.providerId.startsWith('azure'));
-		let accountDropdownOptions = this._azureAccountList.map(account => account.key.accountId);
+		let accountDropdownOptions = this._azureAccountList.map(account => `${account.displayInfo.displayName} - ${account.displayInfo.email ?? ''}`);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/connectionWidget.ts:523]: Don't have the - if there's no e-mail

**Comment by aaomidi** [comment]
Looks like it caught the mock azure accounts not having enough information :D

File: extensions/azurecore/src/test/azureResource/tree/accountTreeNode.test.ts [hunk 3]
```diff
@@ -48,7 +48,8 @@ const mockAccount: azdata.Account = {
 		displayName: 'mock_account@test.com',
 		accountType: 'Microsoft',
 		contextualDisplayName: 'test',
-		userId: 'test@email.com'
+		userId: 'test@email.com',
```
Comment (Charles-Gagnon) [inline extensions/azurecore/src/test/azureResource/tree/accountTreeNode.test.ts:51]: How about giving this an oid (whatever that looks like) instead so it's clear this is supposed to be different?

**Comment by aaomidi** [comment]
For future insider versions lets see how the migration from the old email based key to the new UID based key works.

**Comment by aaomidi** [comment]
If anyone has any ideas on how to make this migration easier, let me know. Maybe automatically removing accounts that have an @ in their ID?

**Pull request merged**